conformance/lang/expressions/string-template-expr/string_template_expr.balt

```diff
@@ -0,0 +1,323 @@
+    string? a4 = "a4";
+    string _ = string `${a4}`; // @error expected '(int|float|decimal|string|boolean)', found 'string?'
+    json a5 = 12;
+    string _ = string `${a5}`; // @error expected '(int|float|decimal|string|boolean)', found 'json'
+    [int, int] a6 = [10, 10];
+    string _ = string `${a6}`; // @error expected '(int|float|decimal|string|boolean)', found '[int,int]'
+}
+
+function f1() returns int {
+    return 10;
+}
+
+Test-Case: error
+Description: Test type of the expression in each interpolation of string-template-expr.
+Labels: string, function-defn, string-template-expr, union-type
+
+type T1 "C"|"S"|map<int>;
+type T2 true|record {};
+type T3 2|3|int[];
+
+function errorFunction() {
+    T1 a1 = "C";
+    string _ = string `${a1}`; // @error expected '(int|float|decimal|string|boolean)', found 'T1'
+    T2  a2 = true;
+    string _ = string `${a2}`; // @error expected '(int|float|decimal|string|boolean)', found 'T2'
+    T3 a3 = 2;
+    string _ = string `${a3}`; // @error expected '(int|float|decimal|string|boolean)', found 'T3'
+}
+
+Test-Case: output
+Description:  Test every character that is not part of an interpolation is interpreted as a literal character.
+Labels: string, function-defn, string-template-expr, int
+
+function init() {
+    string a1 = "a1";
+    string temp1 = string `b1 ${a1}`;
+    io:println(temp1); // @output b1 a1
+    int a2 = 12;
+    string temp2 = string `v1 ${a2} v2`;
+    io:println(temp2); // @output v1 12 v2
+    int a3 = 13;
+    string temp3 = string `$ $ $ ${a2 + a3} $ $ $`;
+    io:println(temp3); // @output $ $ $ 25 $ $ $
+    string temp4 = string `$$${"a2"}$$`;
+    io:println(temp4); // @output $$a2$$
+    string temp5 = string `$v1$${"a2"}$v2$`;
+    io:println(temp5); // @output $v1$a2$v2$
+}
+
+Test-Case: output
+Description:  Test every character that is not part of an interpolation is interpreted as a literal character.
+Labels: string, function-defn, string-template-expr
+
+function init() {
+    string temp1 = string `b1`;
+    io:println(temp1); // @output b1
+    string temp2 = string `v1 v2`;
+    io:println(temp2); // @output v1 v2
+    string temp3 = string `$ $ $ $ $ $`;
+    io:println(temp3); // @output $ $ $ $ $ $
+    string temp4 = string `$$$$`;
+    io:println(temp4); // @output $$$$
+    string temp5 = string `$v1$$v2$`;
+    io:println(temp5); // @output $v1$$v2$
+    string temp6 = string `{}`;
+    io:println(temp6); // @output {}
+}
+
+Test-Case: error
+Description: Test every character that is not part of an interpolation is interpreted as a literal character.
+Fail-Issue: ballerina-platform/ballerina-lang#35515
+Labels: string, function-defn, string-template-expr
+
+function errorFunction() {
+    string _ = string `${`;
+}
+
+function f1() {
+    string name = "Ballerina";
+    string _ = string `${name}`;
+}
+
+function f2() {
+    string _ = string `${`;
+}
+
```

Comment:
Function `f1` and `f2` are unused in the testcase.

Reply:
These two functions are added in `Fail-Issue`